packages/task/src/common/task-protocol.ts

```diff
@@ -187,6 +185,20 @@ export interface TaskInfo {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     readonly [key: string]: any;
 }
+export namespace TaskInfo {
+    export function isAttachableTaskInfo(taskInfo: TaskInfo): taskInfo is AttachableTaskInfo {
+        return typeof taskInfo.terminalId === 'number';
+    }
+}
+
+/**
+ * @todo Use a different mechanism to determine if the task should be attached?
+ *       Reason: Maybe a new task type wants to also be displayed in a terminal.
+ */
+export interface AttachableTaskInfo extends TaskInfo {
+    /** terminal id. Defined if task is run as a terminal process */
+    readonly terminalId: number,
+}
```

Comment:
I don't see the reason for `AttachableTaskInfo`, why wasn't `TaskInfo` sufficient?

Reply:
@vince-fugnitto Actually `AttachableTaskInfo` is not necessary. I'll redo all the changes made to `task-protocol.ts`.